gap/semigroups/semicons.gi

```diff
@@ -805,3 +805,227 @@ for _IsXSemigroup in ["IsTransformationSemigroup",
+          (not IsSubset(semigroups[i],
+                        Range(homomorphisms[i][j]))) then
+        ErrorNoReturn("expected homomorphism from ",
+                      OutNeighbours(D)[i][j],
+                      " to ",
+                      i,
+                      " to have correct source and range");
+      fi;
+    od;
+  od;
+
+  efam := NewFamily("StrongSemilatticeOfSemigroupsElementsFamily", IsSSSE);
+  etype := NewType(efam, IsSSSERep);
+
+  type := NewType(CollectionsFamily(efam),
+                  IsStrongSemilatticeOfSemigroups
+                    and IsComponentObjectRep
+                    and IsAttributeStoringRep);
+
+  # the next section converts the list of homomorphisms into a matrix,
+  # composing when necessary.
+  maps := [];
+  n := Length(semigroups);
+  rtclosure := DigraphReflexiveTransitiveClosure(D);
+  for i in [1 .. n] do
+    Add(maps, []);
+    for j in [1 .. n] do
+      if i = j then
+        # First check that if a homomorphism from i to i was defined, it
+        # is the identity
+        if IsDigraphEdge(D, [i, i]) then
+          if homomorphisms[i][Position(OutNeighboursOfVertex(D, i), i)]
+              <> IdentityMapping(semigroups[i]) then
+             ErrorNoReturn("Expected homomorphism from ",
+                           i, " to ", i,
+                           " to be the identity");
+           fi;
+        fi;
+        Add(maps[i], IdentityMapping(semigroups[i]));
+      elif IsDigraphEdge(rtclosure, [i, j]) then
+        paths        := IteratorOfPaths(D, i, j);
+        path         := NextIterator(paths);
+        len          := Length(path[2]);
+        tobecomposed := List([1 .. len],
+                              x -> homomorphisms[path[1][x]][path[2][x]]);
+        firsthom     := CompositionMapping(tobecomposed);
+
+        # now check the first composition is the same as the composition along
+        # all other paths from i to j
+        while not IsDoneIterator(paths) do
+          path         := NextIterator(paths);
+          len          := Length(path[2]);
+          tobecomposed := List([1 .. len],
+                                x -> homomorphisms[path[1][x]][path[2][x]]);
+          if CompositionMapping(tobecomposed) <> firsthom then
+            ErrorNoReturn("Composing homomorphisms along different paths from ",
+                          i,
+                          " to ",
+                          j,
+                          " does not produce the same result. The ",
+                          "homomorphisms must commute");
+          fi;
+        od;
+        # If no errors so far, then all paths commute and we can add the comp.
+        Add(maps[i], firsthom);
+        # NB. perhaps a dynamic programming approach would be more
+        # efficient here.
+        # for some larger digraphs, the current method might compute some
+        # compositions of homomorphisms several times.
+        # This should be a FIXME.
```

Comment:
Are you going to do this, then?

Reply:
I think it involves dropping use of the `IteratorOfPaths` method and making everything a lot more complicated, and also threatens to take up lots of memory it not done correctly, so for now I've made it a proper FIXME and added more information on what the potential slowdown is. I may try to tackle this over winter but the initialisation still works as-is for now.